service/telemetry.go

```diff
@@ -50,6 +51,10 @@ const AddCollectorVersionTag = true
 const (
 	zapKeyTelemetryAddress = "address"
 	zapKeyTelemetryLevel   = "level"
+
+	// UseOtelForInternalMetricsfeatureGateID is the feature gate ID that controls whether the collector uses open
+	// telemetry for internal metrics.
+	UseOtelForInternalMetricsfeatureGateID = "telemetry.useOtelForInternalMetrics"
```

Comment:
Does this need to be public?

Reply:
Ideally yes. If the user wants to enable the feature they can use the variable directly instead of having constant of their own.

Reply:
The intention is that it is enabled by the flag not code. that is how we envision the use of "featureflags",

Reply:
ok will keep it as private